hommod_rest/services/interpro.py

```diff
@@ -96,99 +99,92 @@ def __init__(self, start, end, ac):
 
 _MAX_INTERPRO_JOBS = 30
 
+
 # Uses interproscan to obtain data: https://code.google.com/p/interproscan/wiki/HowToDownload
 class InterproService (object):
+    def __init__(self, storage_dir=None):
+        self._storage_dir = storage_dir
 
-    def __init__(self):
-
-        self.interproExe = None
-        self.storageDir = None
-        self._njobs = 0
-
-    def _checkinit (self):
-
-        from flask import current_app as flask_app
+    @property
+    def storage_dir(self):
+        return self._storage_dir
 
-#        if not self.interproExe:
-#            self.interproExe = flask_app.config ['INTERPROSCAN']
+    @storage_dir.setter
+    def storage_dir(self, storage_dir):
+        self._storage_dir = storage_dir
 
-        if not self.storageDir:
-            self.storageDir = flask_app.config ['INTERPRODIR']
+    def _checkinit(self):
+        if not self._storage_dir:
+            raise Exception("storage_dir not set")
 
-    def _interpro_lockfile_path (self, ID):
+    def _interpro_lockfile_path(self, ID):
 
-        return os.path.join (self.storageDir, 'lock_%s' % ID)
+        return os.path.join(self.storage_dir, 'lock_%s' % ID)
 
-    def _interpro_count_lockfiles (self):
+    def _interpro_count_lockfiles(self):
 
-        return len (glob (os.path.join (self.storageDir, "lock_*")))
+        return len(glob(os.path.join(self.storage_dir, "lock_*")))
 
     def _create_data_file(self, sequence):
 
-        self._checkinit ()
-        _log.info ("creating interpro file for sequence:\n%s" % sequence)
+        self._checkinit()
+        _log.info("creating interpro file for sequence:\n%s" % sequence)
 
-        if not self.storageDir:
-            raise Exception("storageDir must be set")
+        if not os.path.isdir(self.storage_dir):
+            os.mkdir(self.storage_dir)
 
-        if not os.path.isdir(self.storageDir):
-            os.mkdir(self.storageDir)
+        ID = idForSeq(sequence)  # unique
 
-        ID = idForSeq (sequence) # unique
-
-        outfilepath = os.path.join(self.storageDir, '%s.xml.bz2' % ID)
-        if os.path.isfile (outfilepath):
+        outfilepath = os.path.join(self.storage_dir, '%s.xml.bz2' % ID)
+        if os.path.isfile(outfilepath):
             return outfilepath
 
         # We don't want two processes to build the same interpro file
         # at the same time. So use a lock file:
-        lock = FileLock(self._interpro_lockfile_path (ID))
+        lock = FileLock(self._interpro_lockfile_path(ID))
 
         with lock:
-
             # Wait for a place in line to start an interpro job
-            while self._interpro_count_lockfiles () >= _MAX_INTERPRO_JOBS:
-
-                sleep (10)
+            while self._interpro_count_lockfiles() >= _MAX_INTERPRO_JOBS:
+                sleep(10)
 
             # Wait for the interpro server to finish:
-            jobid = _interpro_submit (sequence)
+            jobid = _interpro_submit(sequence)
             while True:
```

Comment:
If the Interpro service hangs or take a long time, this will become an infinite loop. Please add a timeout. You could also just copy the `InterproService` class from HOPE, but that uses the web service.